contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -62,22 +61,22 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
     // The first key is the blockchain ID of the destination chain, the second key is the message ID, and the value is the info
     // for the uniquely identified message.
     mapping(
-        bytes32 destinationBlockchainID => mapping(uint256 messageID => SentMessageInfo messageInfo)
+        bytes32 destinationBlockchainID => mapping(bytes32 messageID => SentMessageInfo messageInfo)
     ) public sentMessageInfo;
 
     // Tracks the hash of messages that have been received but whose execution has never succeeded.
     // Enables retrying of failed messages with higher gas limits. Message execution is guaranteed to
     // succeed at most once. The first key is the blockchain ID of the sending chain, the second key is the message ID, and
     // the value is the hash of the uniquely identified message whose execution failed.
-    mapping(bytes32 sourceBlockchainID => mapping(uint256 messageID => bytes32 messageHash)) public
+    mapping(bytes32 sourceBlockchainID => mapping(bytes32 messageID => bytes32 messageHash)) public
```

Comment:
Since the messageID encodes the blockchainID, I believe we can simplify this by removing `sourceBlockchainID` as a key, and just keeping the inner map. Same applies to `_relayerRewardAddresses`. However, we need to keep `sentMessageInfo` as is since the messageID does _not_ encode the `destinationBlockchainID`

Reply:
We have to be careful here for a couple of reasons:
- It is technically not correct to say that the message ID "encodes" the source blockchain ID because there is no way to decode what the source blockchain ID value is given a message ID because it's a one way hash.
- While message IDs _should_ be unique from every source blockchain ID now that the source blockchain ID is used as input to that hash that generates the message ID, a single corrupted source subnet can send arbitrary `messageID` values that conflict with other chains. 

An additional step we could take here, is to verify the correctness of the message ID hash in `receiveCrossChainMessage`. We have the source blockchain ID from the warp message itself, and verify that the sender address is always the same as `address(this)` since Teleporter messages are always sent from/to the same address on other chains. The only additional piece of information needed to be able to re-compute the expected message ID hash is the `messageNonce` used by the sender. 

If we have the message include the nonce instead of the message ID, and then re-compute the message ID on the receiving side, then we would be able to rely on its uniqueness across all source chains and could likely leverage that further.

Reply:
These changes are made in #209